databricks_mcp/src/databricks_mcp/connector.py

```diff
@@ -0,0 +1,615 @@
+        )
+        logger.info(f"Successfully created Unity Catalog connection: {connection_name}")
+
+        # Construct workspace URL for viewing the connection
+        workspace_id = w.get_workspace_id()
+        connection_url = f"{w.config.host}/explore/connections/{connection_name}?o={workspace_id}&activeTab=overview"
+        return connection_url
+    except Exception as e:
+        raise RuntimeError(f"Failed to create Unity Catalog connection: {e}") from e
+
+
+# ---------------- Public Entry Point ----------------
+
+
+@experimental
+def register_mcp_server_via_dcr(
+    connection_name: str,
+    mcp_url: str,
+    workspace_client: Optional[WorkspaceClient] = None,
+) -> str:
+    """Register an MCP server via OAuth discovery and Dynamic Client Registration.
+
+    This function performs the complete OAuth discovery and DCR flow:
+    1. Check if Unity Catalog connection already exists (prevents duplicates)
+    2. Discover Protected Resource Metadata (header + well-known fallback)
+    3. Discover Authorization Server Metadata
+    4. Select OAuth scope following MCP spec priority order:
+       a. scope from WWW-Authenticate header (if present)
+       b. scopes_supported from Protected Resource Metadata (if available)
+       c. omit scope parameter (if neither available)
+    5. Perform Dynamic Client Registration with Databricks redirect URI
+    6. Create Unity Catalog connection with the registered client
+
+    Args:
+        connection_name: Name for the Unity Catalog connection
+        mcp_url: The MCP endpoint URL
+        workspace_client: Optional WorkspaceClient instance. If None, a new one is created.
+
+    Returns:
+        The workspace URL to view the connection
+
+    Raises:
+        ValueError: If inputs are invalid
+        RuntimeError: If any step of the discovery or registration process fails
+
+    Example:
+        >>> connection_url = register_mcp_server_via_dcr(
+        ...     connection_name="my_mcp_connection", mcp_url="https://mcp.example.com/api"
+        ... )
+        >>> print(connection_url)
+        'https://workspace.cloud.databricks.com/explore/connections/my_mcp_connection?o=1234567890&activeTab=overview'
+    """
+    if not connection_name:
+        raise ValueError("connection_name cannot be empty")
+    if not mcp_url:
+        raise ValueError("mcp_url cannot be empty")
+
+    logger.info(f"Starting MCP client registration for connection: {connection_name}")
+    logger.info(f"MCP URL: {mcp_url}")
+
+    # Check if connection already exists to prevent duplicates
+    w = workspace_client or WorkspaceClient()
+    try:
+        w.connections.get(connection_name)
+        logger.info(f"Connection '{connection_name}' already exists, skipping DCR")
+        # Construct workspace URL for the existing connection
+        workspace_id = w.get_workspace_id()
+        connection_url = f"{w.config.host}/explore/connections/{connection_name}?o={workspace_id}&activeTab=overview"
```

Comment:
small nit: connection info contains a url can we just return that instead? Same thing in the create_uc_connection above